src/DotLiquid/Tags/Param.cs

```diff
@@ -0,0 +1,112 @@
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Text.RegularExpressions;
+using DotLiquid.Exceptions;
+using DotLiquid.Util;
+
+namespace DotLiquid.Tags
+{
+    /// <summary>
+    /// This tag can be used by Designers to override selected Developer parameters.
+    /// </summary>
+    public class Param : DotLiquid.Tag
+    {
+        private static readonly Regex Syntax = R.B(R.Q(@"([\w\-]+)\s*=\s*(.+)\s*"));
+
+        private delegate void ParamDelegate(Context context, string value);
+
+        private static readonly Dictionary<string, ParamDelegate> Params = new Dictionary<string, ParamDelegate>()
+        {
+            { "culture",     (context, value) => SetCulture(context, value) },
+            { "dateformat",  (context, value) => SetDateFormat(context, value) },
+            { "syntax",      (context, value) => SetSyntax( context, value) },
+            { "using",       (context, value) => AddUsing( context, value) }
+        };
+
+        private ParamDelegate param;
+
+        private string paramValue;
+
+        /// <summary>
+        /// Initializes the Param tag
+        /// </summary>
+        /// <param name="tagName">Name of the parsed tag</param>
+        /// <param name="markup">Markup of the parsed tag</param>
+        /// <param name="tokens">Tokens of the parsed tag</param>
+        /// <exception cref="SyntaxException">If parameter format is invalid.</exception>
+        public override void Initialize(string tagName, string markup, List<string> tokens)
+        {
+            var syntaxMatch = Syntax.Match(markup);
+            if (!syntaxMatch.Success)
+                throw new SyntaxException("Invalid markup format for tag Param: " + markup);
+
+            // Ensure the markup refers to a valid param (discard underscore to support both Ruby and C# naming)
+            var paramName = syntaxMatch.Groups[1].Value.Replace("_", "").ToLower();
+            if (!Params.ContainsKey(paramName))
+                throw new SyntaxException("ParamTagSyntaxException", syntaxMatch.Groups[1].Value, Params.Keys.ToString());
```

Comment:
named parameters would make this more readable as the same input type string is used for all parameters.